src/android/Fingerprint.java

```diff
@@ -71,212 +67,241 @@ public int getValue() {
+                return new Error(
+                        PluginError.BIOMETRIC_HARDWARE_NOT_SUPPORTED.getValue(),
+                        PluginError.BIOMETRIC_HARDWARE_NOT_SUPPORTED.name());
+            case BiometricManager.BIOMETRIC_ERROR_NONE_ENROLLED:
+                return new Error(
+                        PluginError.BIOMETRIC_FINGERPRINT_NOT_ENROLLED.getValue(),
+                        PluginError.BIOMETRIC_FINGERPRINT_NOT_ENROLLED.name());
+            default:
+                return null;
         }
     }
 
-    @Override
-    public void onAuthenticationSuccessful() {
-        Log.e(TAG, "biometric_success");
-        //mCallbackContext.success("biometric_success");
-        sendSuccess("biometric_success");
+    private void showAuthenticationScreen() {
+        KeyguardManager keyguardManager = ContextCompat
+                .getSystemService(cordova.getActivity(), KeyguardManager.class);
+        if (keyguardManager == null
+                || android.os.Build.VERSION.SDK_INT < android.os.Build.VERSION_CODES.LOLLIPOP) {
+            return;
+        }
+        if (keyguardManager.isKeyguardSecure()) {
+            Intent intent = keyguardManager.createConfirmDeviceCredentialIntent(null, null);
+            cordova.setActivityResultCallback(this);
+            cordova.getActivity().startActivityForResult(intent, REQUEST_CODE_CONFIRM_DEVICE_CREDENTIALS);
+        } else {
+            // Show a message that the user hasn't set up a lock screen.
+            sendError(
+                    new Error(
+                            PluginError.BIOMETRIC_SCREEN_GUARD_UNSECURED.getValue(),
+                            "Go to 'Settings -> Security -> Screenlock' to set up a lock screen")
+            );
+        }
     }
 
-    @Override
-    public void onAuthenticationHelp(int helpCode, CharSequence helpString) {
-        Log.e(TAG, "onAuthenticationHelp: " + helpCode + " | " + helpString);
-        sendError(helpCode, helpString != null ? helpString.toString() : "Error has no description.");
-    }
 
     @Override
-    public void onAuthenticationError(int errorCode, CharSequence errString) {
-        switch (errorCode)
-        {
-            case 7:
-            case 9:
-                if(!mDisableBackup) {
-                    showAuthenticationScreen();
-                }else{
-                    sendError(errorCode == 7 ? PluginError.BIOMETRIC_LOCKED_OUT.getValue() : PluginError.BIOMETRIC_LOCKED_OUT_PERMANENT.getValue(), errString.toString());
-                }
-                break;
-
-            default:
-                sendError(errorCode, (errString != null ? errString.toString() : "Error has no description."));
-                break;
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        if (requestCode == REQUEST_CODE_CONFIRM_DEVICE_CREDENTIALS) {
+            if (resultCode == Activity.RESULT_OK) {
+                sendSuccess();
```

Comment:
missing parameter is causing checks to fail ...would you mind looking at the failed checks on the pull request and update accordingly thanks.